### ISSUE TITLE
Cannot execute `Save As...` in git diff panel when failed to save

### ISSUE THREAD

**Issue by ZhangSongyi** [microsoft/vscode#139521]
Issue Type: <b>Bug</b>

1. Create a new folder and initialize an empty git repository in it.
2. Create an empty file in this folder, and  change its properties to `read-only`
3. Open this folder in vscode, go to the `Source Control` panel, and click the untracked file.
4. In the opened git diff panel, type anything in the right side.
5. Try to save the file, which will cause the `Failed to save` pop-up notification.
6. Click the `Save As...` button in the pop-up notification, but nothing happens.

If this file is opened in a normal panel, or in the diff panel when comparing to other files, then the `Save As...` button  in the pop-up notification works properly.

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.22523
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz (8 x 2419)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.79GB (4.35GB free)|
|Process Argv|--crash-reporter-id 99c26a9b-0c96-47a2-825a-93f72f42c876|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscop804:30404766
vscop341:30404997
vsrem710cf:30416617
vscexrecpromp3t3:30407764

```

</details>

**Comment by bpasero** [comment]
Reproduces with normal diff editors too, I will take a look.

**Comment by ZhangSongyi** [comment]
@bpasero  Let me add some experimental phenomena. 
If the read-only file is on the **right side** of the diff editor, things happens exactly the same as I mentioned in the issue.
However, if the file is on the **left side** of the editor, a new panel of this file will be opened in the background when the pop-up notification appears. Then, if we click `Save As...`, then the editor will focus to the new panel and open the `Save As` window successfully.

**Comment by ZhangSongyi** [comment]
@bpasero  Moreover, when comparing a normal file with a read-only file in the diff editor, if we modify the normal file and try to save it (while keep the read-only file unchanged), then the pop-up notification of the read-only file will also shows up, and the following things happens the same to the previous comment.
(Maybe this should marked as a new issue)

**Comment by ZhangSongyi** [comment]
> when comparing a normal file with a read-only file in the diff editor, if we modify the normal file and try to save it (while keep the read-only file unchanged), then the pop-up notification of the read-only file will also show up, and the following things will happen the same to the previous comment.

@bpasero Is this issue also solved in  0cc783e ?

**Comment by bpasero** [comment]
Saving an editor tries to write the contents to disk. So the behaviour is working as intended, even when multiple editors are opened at the same time in one slot.
